app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx

```diff
@@ -369,15 +379,33 @@ export default class SampleViewV2 extends React.Component {
     });
   };
 
-  persistReportOptions = () => {
+  updateHistoryAndPersistOptions = (action = "push") => {
     // save new options to local storage
     const { selectedOptions } = this.state;
+
     // remove exceptions to persistent options
-    const { taxon, ...persistentReportOptions } = selectedOptions;
+    const { taxon, ...localStatePersistedOptions } = selectedOptions;
+
+    const urlState = pick(["view", "selectedOptions"], this.state);
+
+    // Saving on URL enables sharing current view with other users
+    let urlQuery = this.urlParser.stringify(urlState);
+    if (urlQuery) {
+      urlQuery = `?${urlQuery}`;
+    }
+
+    // History state may include some small fields that enable direct loading of previous pages
+    // from browser history without having to request those fields from server (e.g. project)
```

Comment:
Not sure what this means. Unclear what the use case for "push" might be. Can you clarify?

Reply:
It was more of a generic option for whatever we might not want to undo.
But removed for now to not include possible future needs.